packages/@react-stately/table/src/useTableState.ts

```diff
@@ -30,7 +34,10 @@ export interface CollectionBuilderContext<T> {
   columns: Node<T>[]
 }
 
+// TODO: perhaps define things like `items`, `selectedKeys` etc here instead of relying on extend
+// since the table type descriptions for these props should be more specific to table?
```

Comment:
Another open question

Reply:
I mean, you could define it still based on the extend but change the description
```
/** Overriding description */
item?: CollectionBase<T>.item,
```
I *think* something like this will work
maybe it's `CollectionBase<T>['item']`

Reply:
Bleh, the type disappears when I override the description like this:
![image](https://user-images.githubusercontent.com/9661127/118699091-fc0bbc80-b7c5-11eb-9a06-218de443917a.png)

I'll just manually override the types in question for now

Reply:
On second thought, I think I'm ok with the current descriptions, gonna remove these todo comments.